src/gluonts/evaluation/_base.py

```diff
@@ -364,20 +379,29 @@ def get_aggregate_metrics(
             "MSIS": "mean",
         }
 
-        if self.custom_eval_fn is not None:
-            for k, (_, agg_type, _) in self.custom_eval_fn.items():
-                agg_funs.update({k: agg_type})
-
         for quantile in self.quantiles:
             agg_funs[quantile.loss_name] = "sum"
             agg_funs[quantile.coverage_name] = "mean"
 
+        if self.custom_eval_fn is not None:
+            for k, (_, agg_type, _) in self.custom_eval_fn.items():
+                agg_funs.update({k: agg_type})
+
         assert (
             set(metric_per_ts.columns) >= agg_funs.keys()
         ), "Some of the requested item metrics are missing."
 
+        agg_kwargs = {}
+        if self.aggregation_strategy == "inf_only":
+            pass
+        elif self.aggregation_strategy == "all":
+            agg_kwargs = {"skipna": False}
+        elif self.aggregation_strategy == "valid":
+            metric_per_ts = metric_per_ts.apply(np.ma.masked_invalid)
```

Comment:
Do we need an `else` here? Or otherwise you can validate the possible options at construction time.

Reply:
We could validate them there that is true, however the `metric_per_ts.apply(...)` would still need to be run at this point.

Reply:
Moved them to seperate methods instead.